torchvision/transforms/functional_tensor.py

```diff
@@ -105,14 +105,15 @@ def convert_image_dtype(image: torch.Tensor, dtype: torch.dtype = torch.float) -
         return result.to(dtype)
     else:
         input_max = _max_value(image.dtype)
-        output_max = _max_value(dtype)
 
         # int to float
         # TODO: replace with dtype.is_floating_point when torchscript supports it
         if torch.tensor(0, dtype=dtype).is_floating_point():
             image = image.to(dtype)
             return image / input_max
 
+        output_max = _max_value(dtype)
```

Comment:
It's unclear why moving this it fixes the errors. Shall I assume that the if statement is always activated in the problematic cases and thus you never end up executing this line?

Reply:
Yes, the overflow only happens when `dtype` is floating point, so moving this after the conditional solves this